usnistgov/oar-rmm
# integrate Stoplight Elements for rendering RMM OpenAPI JSON file

This PR integrates Stoplight Elements to render the OpenAPI JSON specification of the RMM service.

**Key Changes:**
- Created a new `api.json` file under `src/main/resources/static` containing the OpenAPI schema.
- Created a new `index.html` file under `src/main/resources/static` that integrates the [Stoplight Elements API](https://docs.stoplight.io/docs/elements/d6a8ba3f3c186-stoplight-elements) to render the OpenAPI schema.
- Configured the `index.html` to load and display the contents of the `api.json` using Stoplight Elements.

**How it works: (Source: [Docs](https://docs.stoplight.io/docs/elements/a71d7fcfefcd6-elements-in-html))**
- The following Stoplight Elements JavaScript and CSS scripts are included in the `<head>` tag of the `index.html `file:
```html
<script src="https://unpkg.com/@stoplight/elements/web-components.min.js"></script>
<link rel="stylesheet" href="https://unpkg.com/@stoplight/elements/styles.min.css">
```
- The `elements-api` component from Stoplight Elements is embedded in the `index.html `file.
- The OpenAPI schema from the `api.json` file is passed via `apiDescriptionUrl`.
- When visiting the path `/rmm/index.html`, the API documentation is rendered using Stoplight Elements.



**Steps to test:**
- Start the RMM service.
- Navigate to `http://localhost:8082/rmm/index.html`.
- The API documentation defined in `api.json` should be displayed.

**Notes:**
- The `api.json` was manually edited to include servers for testing the API and to reorganize the APIs in the left panel by changing the order in the `tags` section.
